providers/linux/host_fqdn_integration_linux_test.go

```diff
@@ -39,145 +37,96 @@ const (
+// It will set the container's hostname and domain according to the given
+// values.
+//
+// The container's stdout and stderr are passed through. The test will fail
+// if docker run returns a non-zero exit code.
+func dockerRun(t *testing.T, hostname, domain string, command []string) {
+	t.Helper()
 
-	pwd, err := os.Getwd()
-	if err != nil {
-		t.Fatalf("could not get current directory: %v", err)
+	if _, err := execabs.LookPath("docker"); err != nil {
+		t.Skipf("Skipping because docker was not found: %v", err)
 	}
-	wd := pwd + "../../../"
 
-	reader, err := cli.ImagePull(ctx, cf.Image, types.ImagePullOptions{})
+	// Determine the repository root.
+	_, filename, _, _ := runtime.Caller(0)
+	repoRoot, err := filepath.Abs(filepath.Join(filepath.Dir(filename), "../.."))
 	if err != nil {
-		t.Fatalf("failed to pull image %s: %v", cf.Image, err)
+		t.Fatal(err)
 	}
-	defer reader.Close()
-	io.Copy(os.Stderr, reader)
 
-	resp, err := cli.ContainerCreate(ctx, cf, &container.HostConfig{
-		AutoRemove: false,
-		Binds:      []string{wd + ":/usr/src/elastic/go-sysinfo"},
-	}, nil, nil, "")
-	if err != nil {
-		t.Fatalf("could not create docker conteiner: %v", err)
-	}
-	defer func() {
-		err = cli.ContainerRemove(ctx, resp.ID, types.ContainerRemoveOptions{
-			Force: true, RemoveVolumes: true,
-		})
-		if err != nil {
-			t.Logf("WARNING: could not remove docker container: %v", err)
-		}
-	}()
+	// Use the same version of Go inside the container.
+	goVersion := strings.TrimPrefix(runtime.Version(), "go")
 
-	if err := cli.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
-		t.Fatalf("could not start docker container: %v", err)
+	args := []string{
+		"run",
+		"--rm",
+		"-v", build.Default.GOPATH + ":/go", // Mount GOPATH for caching.
+		"-v", repoRoot + ":/go-sysinfo",
+		"-w=/go-sysinfo",
 	}
-
-	statusCh, errCh := cli.ContainerWait(ctx, resp.ID, container.WaitConditionNotRunning)
-	select {
-	case err := <-errCh:
-		if err != nil {
-			// Not using fatal as we might be able to recover the container
-			// logs.
-			t.Errorf("docker ContainerWait failed: %v", err)
-		}
-	case s := <-statusCh:
-		if s.StatusCode != 0 {
-			msg := fmt.Sprintf("container exited with status code %d", s.StatusCode)
-			if s.Error != nil {
-				msg = fmt.Sprintf("%s: error: %s", msg, s.Error.Message)
-			}
-
-			// Not using fatal as we might be able to recover the container
-			// logs.
-			t.Errorf(msg)
-		}
+	if hostname != "" {
+		args = append(args, "--hostname="+hostname)
 	}
-
-	out, err := cli.ContainerLogs(ctx, resp.ID, types.ContainerLogsOptions{ShowStderr: true, ShowStdout: true})
-	if err != nil {
-		t.Fatalf("could not get container logs: %v", err)
+	if domain != "" {
+		args = append(args, "--domainname="+domain)
+	}
+	args = append(args, "golang:"+goVersion)
+	args = append(args, command...)
+
+	cmd := execabs.Command("docker", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
```

Comment:
Capture these in a buffer? Then they can be emitted in the `t.Logf` call.

Reply:
SGTM. Here's the new output:

```
ubuntu@neat-chimpanzee:/go-sysinfo/providers/linux$ go test  -run TestHost_FQDN -tags integration -v .
=== RUN   TestHost_FQDN
=== RUN   TestHost_FQDN/TestHost_FQDN_set_hostname+domainname
    host_fqdn_integration_linux_test.go:84: Running docker container using ["run" "--rm" "-v" "/home/ubuntu/go:/go" "-v" "/go-sysinfo:/go-sysinfo" "-w=/go-sysinfo" "--hostname=hostname" "--domainname=some.domain" "golang:1.22.2" "go" "test" "-v" "-tags" "integration,docker" "-run" "^TestHost_FQDN_set$" "./providers/linux"]
    host_fqdn_integration_linux_test.go:84: Container output:
        === RUN   TestHost_FQDN_set
        --- PASS: TestHost_FQDN_set (0.00s)
        PASS
        ok      github.com/elastic/go-sysinfo/providers/linux   0.005s
    host_fqdn_integration_linux_test.go:84: Exiting container
=== RUN   TestHost_FQDN/TestHost_FQDN_set_hostname_only
    host_fqdn_integration_linux_test.go:84: Running docker container using ["run" "--rm" "-v" "/home/ubuntu/go:/go" "-v" "/go-sysinfo:/go-sysinfo" "-w=/go-sysinfo" "--hostname=hostname.some.domain" "golang:1.22.2" "go" "test" "-v" "-tags" "integration,docker" "-run" "^TestHost_FQDN_set$" "./providers/linux"]
    host_fqdn_integration_linux_test.go:84: Container output:
        === RUN   TestHost_FQDN_set
        --- PASS: TestHost_FQDN_set (0.00s)
        PASS
        ok      github.com/elastic/go-sysinfo/providers/linux   0.006s
    host_fqdn_integration_linux_test.go:84: Exiting container
=== RUN   TestHost_FQDN/TestHost_FQDN_not_set
    host_fqdn_integration_linux_test.go:84: Running docker container using ["run" "--rm" "-v" "/home/ubuntu/go:/go" "-v" "/go-sysinfo:/go-sysinfo" "-w=/go-sysinfo" "golang:1.22.2" "go" "test" "-v" "-count" "1" "-tags" "integration,docker" "-run" "^TestHost_FQDN_not_set$" "./providers/linux"]
    host_fqdn_integration_linux_test.go:84: Container output:
        === RUN   TestHost_FQDN_not_set
        --- PASS: TestHost_FQDN_not_set (0.00s)
        PASS
        ok      github.com/elastic/go-sysinfo/providers/linux   0.007s
    host_fqdn_integration_linux_test.go:84: Exiting container
--- PASS: TestHost_FQDN (54.56s)
    --- PASS: TestHost_FQDN/TestHost_FQDN_set_hostname+domainname (17.82s)
    --- PASS: TestHost_FQDN/TestHost_FQDN_set_hostname_only (18.14s)
    --- PASS: TestHost_FQDN/TestHost_FQDN_not_set (18.60s)
PASS
ok      github.com/elastic/go-sysinfo/providers/linux   54.562s
```